api-controllers/FabrikBaseController.js

```diff
@@ -63,7 +63,7 @@ class FabrikBaseController extends BaseController {
             resourceId: req.params.instance_id,
             operation: operationType
           }
-        });
+        }, plan);
```

Comment:
not related to this PR but in general for SF2.0, during Bind we dont acquire any READ locks. Shouldn't we be acquiring it? look here `middleware.checkBlockingOperationInProgress` - if we dont see a write lock we just proceed, but what if someone fires update while a bind is in progress? or say fires a delete ?

Reply:
As discussed 'async_ops_supporting_parallel_sync_ops' parameter in plan having the info.

Reply:
Maybe I didn't get this correctly , someone firing update or delete while bind is in progress cannot happen as CF would not allow it , correct ? Because for user triggered update/delete , the request will come via CF ?!

I see that now since we have decoupled outdated instance update from CF , there is a chance that while bind is in progress an outdated instance update can go through.

Is this understanding correct ?

Reply:
@kamath-prasad - Your understanding is correct. In summary this is what we are doing:
1. For SYNCH operations no locks are acquired by SF.
2. Only for ASYCNH operations we acquire lock and we flag that lock as READ/WRITE. If a LOCK is flagged as READ, then it would allow for parallel SYNCH requests to go through.

So we still rely on the platform for locking SYNCH operations as you have rightly pointed out and this is sth if K8s does not provide similar to CF, then we must revisit this model and have our locking completely independent of platform. Hope this clarifies any further doubts that you had.

Reply:
Adding to what @jagadish-kb said, with scheduled update not going through CF, even if it happens that update comes while bind in progress and update brings the VM down, the retry would ensure that the credential is created.